Python/scoreP2B.py

```diff
@@ -45,18 +39,57 @@ def scoreP2BImage(truthPath, testPath):
 
 
 def scoreP2B(truthDir, testDir):
-    # Iterate over each file and call scoring executable on the pair
     scores = []
+    features = defaultdict(list)
+
+    # Verify that test filenames match truth filenames
     for truthFile in sorted(os.listdir(truthDir)):
         testPath = matchFeatureInputFile(truthFile, testDir)
         truthPath = os.path.join(truthDir, truthFile)
 
         # truthFile ~= 'ISIC_0000003_globules.png'
         datasetName = os.path.splitext(truthFile)[0]
-        metrics = scoreP2BImage(truthPath, testPath)
+
+        featureName = datasetName.split('_')[2]
+        features[featureName].append((truthPath, testPath))
+
+    # Dimensions of resized images to add to concatenated arrays
+    width = 512
+    height = 512
+
+    for featureName, paths in features.viewitems():
+        # Pre-allocate concatenated arrays
+        numFiles = len(paths)
+        truthArray = np.zeros(numFiles * width * height, np.uint8)
+        testArray = np.zeros(numFiles * width * height, np.uint8)
+
+        for index, (truthPath, testPath) in enumerate(paths):
+            # Load images and verify that dimensions match
+            truthImage = loadSegmentationImage(truthPath)
+            testImage = loadSegmentationImage(testPath)
+
+            if testImage.size != truthImage.size:
+                raise ScoreException('Image %s has dimensions %s; expected %s.' %
+                                     (os.path.basename(testPath), testImage.size,
+                                      truthImage.size))
+
+            # Resize images
+            truthImage = resizeImage(truthImage, width, height)
+            testImage = resizeImage(testImage, width, height)
+
+            # Convert images
+            truthImage = convertToNumPyArray(truthImage)
+            testImage = convertToNumPyArray(testImage)
+
```

Comment:
Then do the thresholding with `> 128` here (it doesn't matter if it's before or after `.flatten()`.